### ISSUE TITLE
`Content-Type` must be present in `multipart/form-data`

### ISSUE THREAD

**Issue by SuperFluffy** [actix/actix-web#1947]
## Expected Behavior

`actix_multipart::Field::content_disposition` should return `&ContentDisposition`, because the `ContentDisposition` is mandatory on multipart form subparts.

## Current Behavior

At the moment, we have the following function signature:`actix_multipart::Field::content_disposition(&self) -> Option<ContentDisposition>`.

Interestingly, the code actually notes this: https://github.com/actix/actix-web/blob/master/actix-multipart/src/server.rs#L417-L426:

```rust
    /// Get the content disposition of the field, if it exists
    pub fn content_disposition(&self) -> Option<ContentDisposition> {
        // RFC 7578: 'Each part MUST contain a Content-Disposition header field
        // where the disposition type is "form-data".'
        if let Some(content_disposition) = self.headers.get(&header::CONTENT_DISPOSITION)
        {
            ContentDisposition::from_raw(content_disposition).ok()
        } else {
            None
        }
    }
```

## Possible Solution

`Field`s are constructed by `multipart.try_next().await`. A `Field` should be a valid subpart of the multipart stream, so `try_next` should fail if it is not.

## Context

RFC 7578 states the following in https://tools.ietf.org/html/rfc7578#section-4.2:

> Each part MUST contain a Content-Disposition header field [RFC2183]
   where the disposition type is "form-data".  The Content-Disposition
   header field MUST also contain an additional parameter of "name";

So modelling `Field::content_disposition` as an `Option` is actually too lenient.

* Actix Web Version: 3.0

**Comment by craigpastro** [comment]
Hello 👋 Does anybody mind if I give this one a shot?

**Comment by roman-holovin** [comment]
`multipart/byteranges` message is not required to have `Content-Disposition`. It is a `multipart/form-data` feature. 

https://datatracker.ietf.org/doc/html/rfc7233#appendix-A

**Comment by SuperFluffy** [comment]
So many standards.

I wonder if it makes sense to represent multipart form-data and byterange on the typelevel, together with an escape hatch for other types (are the any?), and then tie content disposition to these types.

Would require quite the refactoring though and would probably be a breaking change.

**Comment by robjtede** [comment]
closed in #2451
